### PR TITLE
Improve assertions

### PR DESCRIPTION
**Pull request by carusogabriel**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Following #25420 in `3.4` branch.

Thanks to [`Rector`](https://github.com/rectorphp/rector) that helped me this time :smile:

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Submitting anything on 4.0 should likely go to 3.4 instead.

File: src/Symfony/Component/DependencyInjection/Tests/ContainerBuilderTest.php [hunk 1]
```diff
@@ -358,7 +358,7 @@ public function testCreateProxyWithRealServiceInstantiator()
         $foo1 = $builder->get('foo1');
 
         $this->assertSame($foo1, $builder->get('foo1'), 'The same proxy is retrieved on multiple subsequent calls');
-        $this->assertSame('Bar\FooClass', get_class($foo1));
+        $this->assertInstanceOf('Bar\FooClass', $foo1);
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Tests/ContainerBuilderTest.php]: this is not the same assertion (it now allows subclasses)

Comment (carusogabriel) [reply src/Symfony/Component/DependencyInjection/Tests/ContainerBuilderTest.php]: Oh, thanks :smile:

File: src/Symfony/Bridge/Doctrine/Tests/ManagerRegistryTest.php [hunk 2]
```diff
@@ -35,7 +35,7 @@ public function testResetService()
 
         $foo = $container->get('foo');
         $foo->bar = 123;
-        $this->assertTrue(isset($foo->bar));
+        $this->assertObjectHasAttribute('bar', $foo);
```
Comment (ro0NL) [inline src/Symfony/Bridge/Doctrine/Tests/ManagerRegistryTest.php]: different assertion

File: src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php [hunk 3]
```diff
@@ -165,7 +165,7 @@ public function testLoadServices()
         $this->assertArrayHasKey('another_alias_for_foo', $aliases);
         $this->assertEquals('foo', (string) $aliases['another_alias_for_foo']);
         $this->assertFalse($aliases['another_alias_for_foo']->isPublic());
-        $this->assertTrue(isset($aliases['another_third_alias_for_foo']));
+        $this->assertArrayHasKey('another_third_alias_for_foo', $aliases);
```
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php]: different assertion (now allows null)

**Comment by carusogabriel** [comment]
@nicolas-grekas Appointments from @stof and @ro0NL sound valid. Should we revert #25420 and similaris?

**Comment by nicolas-grekas** [comment]
Please review and revert what should be yes.

**Comment by sroze** [comment]
Tbh, a fair amount of similar pull-requests have been refused already because the value they bring vs the time they consume is questionable :)

**Comment by ro0NL** [comment]
Agree.. but :+1: for changing to `assertNull, assertCount, assertTrue` for the obvious assertions. If that's on someone's list to do :)

**Comment by carusogabriel** [comment]
Closing as it will need some work :construction:

**Pull request closed without merging**